### PR TITLE
Add `cargo-0.60.0` as a new benchmark.

### PR DESCRIPTION
**Pull request by nnethercote**
I obtained the code with:
```
target/release/collector download -c primary crate cargo 0.60.0
```

`cargo-0.60.0/perf-config.json` and `cargo-0.60.0/0-println.patch` are
based on the corresponding files in `cargo/`.

`cargo-0.60.0` is quite a bit slower to compile than `cargo`. Times for
`full` builds of the final crate on my machine:
```
		cargo   cargo-0.60.0
cargo check 	4.26    6.92
cargo debug 	10.13   19.72
cargo opt 	17.59   35.67
```
And to run the entire benchmark (including compiling all crates):
```
cargo:        Check  real 18.46 user  84.04 sys 18.01
cargo-0.60.0: Check  real 18.16 user 100.21 sys 13.45
cargo:        Debug  real 25.98 user 163.99 sys 21.99
cargo-0.60.0: Debug  real 32.92 user 205.82 sys 18.37
cargo:        Opt    real 38.28 user 450.19 sys 20.37
cargo-0.60.0: Opt    real 55.54 user 585.34 sys 19.06
```

`cargo` will be removed soon.

### CODE REVIEW

**Review by Mark-Simulacrum — APPROVED** [review]
This looks good to me (I didn't try to confirm this exactly reproduces Cargo, but seems OK -- we may, I suppose, want to move to some kind of crates.io download + set of patches system to avoid inlining a bunch of raw files, but that has its own annoyances).

It looks like it needs a rebase and I'd like to hold off on merging until we make sure that we have some spare time in our builder (after the style-servo removal I just merged).

**Comment by nnethercote** [comment]
I have rebased.

**Review by rylev — APPROVED** [review]

**Comment by rylev** [comment]
Going to merge since I think the builder has had time to recover.

**Pull request merged**
